src/mallocTracer.cpp

```diff
@@ -153,21 +154,22 @@ void MallocTracer::patchLibraries() {
 
     while (_patched_libs < native_lib_count) {
         CodeCache* cc = (*native_libs)[_patched_libs++];
+        UnloadProtection handle = UnloadProtection(cc);
 
-        cc->patchImport(im_malloc, (void*)malloc_hook);
-        cc->patchImport(im_realloc, (void*)realloc_hook);
-        cc->patchImport(im_free, (void*)free_hook);
-        cc->patchImport(im_aligned_alloc, (void*)aligned_alloc_hook);
+        handle.patchImport(im_malloc, (void*)malloc_hook);
```

Comment:
We should still use `cc->patchImport`. The fact that `UnloadProtection` is alive, should give us the needed protection.

Reply:
What I'm trying to achieve here is that you cannot patch without having an active `UnloadProtection`